tests/bluetooth/bsim_bt/bsim_test_audio/src/mics_test.c

```diff
@@ -0,0 +1,494 @@
+	}
+}
+
+static void aics_status_cb(struct bt_conn *conn, struct bt_aics *inst, int err,
+			   bool active)
+{
+	if (err != 0) {
+		FAIL("AICS status cb err (%d)", err);
+		return;
+	}
+
+	g_aics_active = active;
+
+	if (conn == NULL) {
+		g_cb = true;
+	}
+}
+
+static void aics_description_cb(struct bt_conn *conn, struct bt_aics *inst,
+				int err, char *description)
+{
+	if (err != 0) {
+		FAIL("AICS description cb err (%d)", err);
+		return;
+	}
+
+
+	strncpy(g_aics_desc, description, sizeof(g_aics_desc) - 1);
+	g_aics_desc[sizeof(g_aics_desc) - 1] = '\0';
+
+	if (conn == NULL) {
+		g_cb = true;
+	}
+}
+
+static struct bt_mics_cb mics_cb = {
+	.mute = mics_mute_cb,
+};
+
+static struct bt_aics_cb aics_cb = {
+	.state = aics_state_cb,
+	.gain_setting = aics_gain_setting_cb,
+	.type = aics_input_type_cb,
+	.status = aics_status_cb,
+	.description = aics_description_cb
+};
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (err != 0) {
+		FAIL("Failed to connect to %s (%u)\n", addr, err);
+		return;
+	}
+
+	printk("Connected to %s\n", addr);
+	g_conn = conn;
+	g_is_connected = true;
+}
+
+static struct bt_conn_cb conn_callbacks = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+static int test_aics_standalone(void)
+{
+	int err;
+	int8_t expected_gain;
+	uint8_t expected_input_mute;
+	uint8_t expected_mode;
+	uint8_t expected_input_type;
+	bool expected_aics_active;
+	char expected_aics_desc[AICS_DESC_SIZE];
+
+	printk("Deactivating AICS\n");
+	expected_aics_active = false;
+	err = bt_mics_aics_deactivate(mics.aics[0]);
+	if (err != 0) {
+		FAIL("Could not deactivate AICS (err %d)\n", err);
+		return err;
+	}
+	WAIT_FOR(expected_aics_active == g_aics_active);
+	printk("AICS deactivated\n");
+
+	printk("Activating AICS\n");
+	expected_aics_active = true;
+	err = bt_mics_aics_activate(mics.aics[0]);
+	if (err != 0) {
+		FAIL("Could not activate AICS (err %d)\n", err);
+		return err;
+	}
+	WAIT_FOR(expected_aics_active == g_aics_active);
+	printk("AICS activated\n");
+
+	printk("Getting AICS state\n");
+	g_cb = false;
+	err = bt_mics_aics_state_get(NULL, mics.aics[0]);
+	if (err != 0) {
+		FAIL("Could not get AICS state (err %d)\n", err);
+		return err;
+	}
+	WAIT_FOR(g_cb);
+	printk("AICS state get\n");
+
+	printk("Getting AICS gain setting\n");
+	g_cb = false;
+	err = bt_mics_aics_gain_setting_get(NULL, mics.aics[0]);
+	if (err != 0) {
+		FAIL("Could not get AICS gain setting (err %d)\n", err);
+		return err;
+	}
+	WAIT_FOR(g_cb);
+	printk("AICS gain setting get\n");
+
+	printk("Getting AICS input type\n");
+	expected_input_type = BT_AICS_INPUT_TYPE_UNSPECIFIED;
+	err = bt_mics_aics_type_get(NULL, mics.aics[0]);
+	if (err != 0) {
+		FAIL("Could not get AICS input type (err %d)\n", err);
+		return err;
+	}
+	/* Expect and wait for input_type from init */
+	WAIT_FOR(expected_input_type == g_aics_input_type);
```

Comment:
The callback sets g_cb, but it is not being used.
(It does not have to be used, either, but that kind of defeats a possible purpose of ruling out the callback being called for a client.)

Reply:
I don't disagree with you there (I also feel like we've had this comment before, and I guess it was only fixed for something else as the issue also appears in the VCS AICS test).

Reply:
It is not a big issue - do as you find suitable

Reply:
Was fixed though :)